fauna/client/client.py

```diff
@@ -179,6 +180,35 @@ def get_query_timeout(self) -> Optional[timedelta]:
         else:
             return None
 
+    def paginate(
+        self,
+        fql: Query,
+        opts: Optional[QueryOptions] = None,
+    ) -> Generator[Page, None, None]:
+        """
+        Run a query on Fauna and returning an iterator of results. If the query
+        returns a Page, the iterator will fetch additional Pages, until 
+
+        :param fql: A string, but will eventually be a query expression.
+        :param opts: (Optional) Query Options
+
+        :return: a :class:`QueryResponse`
+
+        :raises NetworkError: HTTP Request failed in transit
+        :raises ProtocolError: HTTP error not from Fauna
+        :raises ServiceError: Fauna returned an error
+        :raises ValueError: Encoding and decoding errors
+        :raises TypeError: Invalid param types
+        """
+
+        if not isinstance(fql, Query):
+            err_msg = f"'fql' must be a Query but was a {type(fql)}. You can build a " \
+                       f"Query by calling fauna.fql()"
+            raise TypeError(err_msg)
+        
+        from fauna.query.iterator import QueryIterator
```

Comment:
I had trouble with circular dependencies, so this was one way to get around that. I'm not positive I put the new class in a reasonable place considering. Thoughts on this?

Reply:
Hmm, I agree with your choice logically, but let's bundle it under client/ to avoid the circular dep.

Reply:
done!